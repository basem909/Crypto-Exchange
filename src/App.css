@import url("https://fonts.googleapis.com/css2?family=Lato&family=Special+Elite&display=swap");
@import url("http://fonts.cdnfonts.com/css/gill-sans");
