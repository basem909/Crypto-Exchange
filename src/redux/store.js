import {
  legacy_createStore as legacyCreateStore,
  combineReducers,
  applyMiddleware,
} from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import showCoins from './reducer';
import { showCoinList } from './action-creators';

const allReducers = combineReducers({
  showCoins,
});
const store = legacyCreateStore(allReducers, applyMiddleware(logger, thunk));
store.dispatch(showCoinList());

export default store;
