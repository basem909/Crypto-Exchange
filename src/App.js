import { useDispatch } from 'react-redux';
import { Routes, Route } from 'react-router-dom';
import './App.css';
import CoinDetails from './components/coinDetails';
import CoinsExihbit from './components/coinList';
import {
  searchCoinlist,
  showDetails,
} from './redux/action-creators';

const App = () => {
  const dispatch = useDispatch();
  const clickHandler = (id) => {
    dispatch(showDetails(id));
  };
  const changehandler = (str) => {
    dispatch(searchCoinlist(str));
  };
  return (
    <div className="App">
      <Routes>
        <Route
          path="/"
          element={(
            <CoinsExihbit
              clickHandler={clickHandler}
              changeHandler={changehandler}
            />
          )}
        />
        <Route
          path="/coinDetails"
          element={<CoinDetails clickHandler={clickHandler} />}
        />
      </Routes>
    </div>
  );
};

export default App;
