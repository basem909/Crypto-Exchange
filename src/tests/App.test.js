import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App';
import store from '../redux/store';

describe('App snapshot', () => {
  afterEach(cleanup);

  it('should render App correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
