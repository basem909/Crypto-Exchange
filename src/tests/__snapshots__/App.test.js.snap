// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should render App correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <header
        class="header"
      >
        <h1
          class="title"
        >
          Crypto World
        </h1>
      </header>
      <img
        alt="crypto"
        class="background"
        src="B.jpeg"
      />
      <form
        class="search-bar"
      >
        <span
          name="search"
        >
           Search by Coin here 
        </span>
        <input
          id="searchBar"
          name="search"
          placeholder="Coin name or symbol"
          type="text"
        />
      </form>
      <div
        class="coins-container"
      />
    </div>
  </div>
</DocumentFragment>
`;
